### PR TITLE
UnitRemoveBuffs, UnitRemoveBuffsEx, UnitHasBuffsEx, UnitCountBuffsEx

### PR DESCRIPTION
**Pull request by WaterKnight**

### CODE REVIEW

File: common.j [hunk 1]
```diff
@@ -14630,18 +14630,124 @@ This native is used to keep abilities when morphing units.
...
+@note Buffs are either positive or negative. Buffs are either magical or physical or neither. Buffs are either timed life (see `UnitApplyTimedLife`) or not.
+Buffs either stem from an aura or not. Buffs are either dispelable (via dispelling abilities) or not.
+
+@note Specifying both `removePositive` and `removeNegative` as true will include both positive and negative buffs (given that other criteria are satisfied).
+These two criteria are additive.
+
+@note Specifying both `magic` and `physical` as true will not include any buff.
+Specifying `magic` as true and `physical` as false will include only magical buffs (given that other criteria are satisfied), no physical buffs nor buffs
+that are neither magical nor physical.
+Specifying `magic` as false and `magic` as true will include only physical buffs (given that other criteria are satisfied), no magical buffs nor buffs
```
Comment (lep) [inline common.j:14665]: `magic` as false and true?

File: common.j [hunk 2]
```diff
@@ -14630,18 +14630,124 @@ This native is used to keep abilities when morphing units.
...
+Buffs either stem from an aura or not. Buffs are either dispelable (via dispelling abilities) or not.
+
+@note Specifying both `removePositive` and `removeNegative` as false will include both positive and negative buffs (given that other criteria are satisfied).
+These two criteria are additive; thus specifying both `removePositive` and `removeNegative` will include both positve and negative buffs as well
+(given that other criteria are satisfied).
+
+@note Specifying both `magic` and `physical` as false will include both magical and physical buffs (given that other criteria are satisfied).
+Specifying `magic` as true and `physical` as false will include only magical buffs (given that other criteria are satisfied), no physical buffs nor buffs
+that are neither magical nor physical.
+Specifying `magic` as false and `magic` as true will include only physical buffs (given that other criteria are satisfied), no magical buffs nor buffs
```
Comment (lep) [inline common.j:14725]: c'n'p error like above?

File: common.j (hunk 1 above)
Comment (WaterKnight) [reply common.j]: fixed

File: common.j (hunk 2 above)
Comment (WaterKnight) [reply common.j]: fixed

**Pull request merged**